### ISSUE TITLE
ansible 2.1.0, s3 module bug

### ISSUE THREAD

**Issue by alxschwarz** [ansible/ansible-modules-core#3758]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

s3 module
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

Only default settings
##### OS / ENVIRONMENT

I think N/A. But we use Ubuntu:

```
NAME="Ubuntu"
VERSION="14.04.4 LTS, Trusty Tahr"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 14.04.4 LTS"
VERSION_ID="14.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
```
##### SUMMARY

Error during using s3 module. For example, during get file from s3 bucket.
##### STEPS TO REPRODUCE

Run playbook

```

---
- name: Test playbook
  hosts: localhost
  tasks:
    - name: Get file from S3
      s3: bucket=<bucket_name> object=/path/to/file> dest=/tmp/file mode=get
```
##### EXPECTED RESULTS

File to /tmp directory
##### ACTUAL RESULTS

Error during ansible task running.

```
TASK [Get file from S3] ************************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to connect to S3: Region  does not seem to be available for aws module boto.s3. If the region definitely exists, you may need to upgrade boto or extend with endpoints_path"}
```

**Comment by nemd** [comment]
any update regarding this bug ?

**Comment by rohitkothari** [comment]
I think this has already been resolved here. I pulled the change from this [commit](https://github.com/ansible/ansible-modules-core/pull/3347/commits/ec0f2113e05fd4435816189d8c7da9c8d5d4d42c) into my ansible's s3.py (Ansible version - 2.1.0.0) and was able to surpass the above error.

Here is the pull request and the discussion behind [this](https://github.com/ansible/ansible-modules-core/pull/3347).

**Comment by jmvbxx** [comment]
As of today this error still occurs with most recent versions of apt-get & pip ansible

**Comment by Juberstine** [comment]
Same issue as well in 2.1.0.0. Installed via Brew

**Comment by listx** [comment]
@rohitkothari I applied the same patch as well to my local s3.py file (ansible 2.1.0.0 via brew) and got it working. The fix has already been merged into `devel` branch; we should probably close this issue.

**Comment by ssbarnea** [comment]
No way, as long this bug is still present under current release it is not fixed. I doubt merge into devel would count as a proper definition of "done" ;)

A bug is done when is does reach production / release, not the development branch.

I am really displeased to see that such a critical bug ended up into a release and that it was not hot-fixed yet after almost 3 months.

Clearly Ansible would need some better QA process that would validate that basic functionality is not messed across new releases.

**Comment by listx** [comment]
Does Ansible even have a hotfix process? I am new to this community and don't know any better than to use devel (rc) versions.

**Comment by emalloy** [comment]
strangely enough i only get this error in us-east-1

**Comment by TaiSHiNet** [comment]
Isn't this a duplicate of #3447 ?
If so, follow the issue on a single thread please

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by alxschwarz** [comment]
I think bug is fixed in version v2.1.1.0. Just tested.
Who can check also?

**Comment by chepins** [comment]
Please could you merge it to master?

I changed manually and works fine!

**Comment by TaiSHiNet** [comment]
@chepins afaik this was fixed in 2.1.1.0 which would mean it's fixed on master. 
If you're still seeing issues please append some info or open a new issue
